### PR TITLE
Final step

### PR DESCRIPTION
**Pull request by Said-laasri**
**Hello react rails**

- Create a new Rails app called 'hello-rails-react'.
- Use the commend rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.
- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/controllers/api/v1/greetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Api::V1::GreetingsController < ApiController
+  def index
+    @greeting = Greeting.order(Arel.sql('RANDOM()')).first
+  end
+end
```
Comment (HAMMAS-SALEEM) [inline app/controllers/api/v1/greetings_controller.rb:5]: image

- [ ] Your application is working on the browser but when it comes to that specific url then it shows this errors shown in the image above :arrow_up:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Said-laasri :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the reqirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/greetings_controller.rb (hunk 1 above)
Comment (Said-laasri) [reply app/controllers/api/v1/greetings_controller.rb:5]: @HAMMAS-SALEEM thanks for you review all working 
Screenshot from 2022-11-21 19-00-49

**Comment by Said-laasri** [comment]
@HAMMAS-SALEEM   all good on the server 

Screenshot from 2022-11-21 19-02-03


Screenshot from 2022-11-21 19-01-58

**Review by julie-ify — APPROVED** [review]
Hi @Said-laasri,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Said-laasri** [comment]
@julie-ify thanks for approving my project

**Pull request merged**